webui/react/src/App.tsx

```diff
@@ -18,7 +18,7 @@ import appRoutes from 'routes';
 import { correctViewportHeight, refreshPage } from 'utils/browser';
 
 import css from './App.module.scss';
-import { paths } from './routes/utils';
+import { checkServerAlive, paths, serverAddress } from './routes/utils';
```

Comment:
**chore fix**: Not introduced by the PR but I noticed in `checkServerAlive` there is a small logic that needs to be fixed. This portion needs to be re-written:
```
    attrs.forEach(att => {
      if (!(att in data)) {
        // the server doesn't look like a determined server
        return false;
      }
    });
```
to something like:
```
    for (let attr of attrs) {
      // The server doesn't look like a determined server.
      if (!(attr in data)) return false;
    }
```
The return call in `forEach` doesn't break out of the callback to exit the overall `checkServerAlive` function with a return value. Changing it to `for` should do the trick.

Reply:
aah good catch!

Reply:
**questions:** couple of questions:
1. Can `checkServerAlive` use `getInfo` already defined in `api.ts`?
2. will `checkServerAlive` be used outside of `App`? Does it need to be reusable vs just having it exist in `App.ts`? (in case referencing `getInfo` causes a circular dependency)

Reply:
1. we should be able to. that's a good suggestion. I wanted to be able to quickly change the server address since it's originally used to check whether a given remote server is valid and alive or not. should still be doable when using `getInfo`

2. it currently is used outside of `App` in the omnibar when the user provides a new server address to switch to. I'll watch out for that circular dependency